pkg/internal/util/kubeconfig/paths.go

```diff
@@ -91,3 +92,76 @@ func discardEmptyAndDuplicates(paths []string) []string {
 	}
 	return paths[:kept]
 }
+
+// homeDir returns the home directory for the current user.
+// On Windows:
+// 1. the first of %HOME%, %HOMEDRIVE%%HOMEPATH%, %USERPROFILE% containing a `.kube\config` file is returned.
+// 2. if none of those locations contain a `.kube\config` file, the first of %HOME%, %USERPROFILE%, %HOMEDRIVE%%HOMEPATH% that exists and is writeable is returned.
+// 3. if none of those locations are writeable, the first of %HOME%, %USERPROFILE%, %HOMEDRIVE%%HOMEPATH% that exists is returned.
+// 4. if none of those locations exists, the first of %HOME%, %USERPROFILE%, %HOMEDRIVE%%HOMEPATH% that is set is returned.
+// NOTE this is from client-go. Rather than pull in client-go for this one
+// standalone method, we have a fork here.
+// https://github.com/kubernetes/client-go/blob/6d7018244d72350e2e8c4a19ccdbe4c8083a9143/util/homedir/homedir.go
+// We've modified this to require injecting os.Getenv and runtime.GOOS as a dependencies for testing purposes
```

Comment:
i think there were complains that this logic in client-go is not ideal, but i don't remember the PR/issue or any details.

Reply:
so this seems fine as a fork in kind.

Reply:
This 100% matches logic wise, we want to produce the same path. We already had this code, it just moved and got more testable.

Reply:
FWIW the client-go logic is now pretty well defined to be $HOME, just with fallbacks on windows. This is locally internal because it's only the "homedir" for the purposes of kubeconfig. Otherwise we'd ask the go stdlib which added this in 1.12 https://golang.org/pkg/os/#UserHomeDir